### PR TITLE
Makes call to the server to save a coach

### PR DESCRIPTION
**Pull request by viniciusffj**
PR for issue #2.

Some notes:
- We still don't have a concrete code style for the whole repo, for example some files had 2 spaces indentations and others 4 spaces. So I tried to apply the [Style Guide](http://elm-lang.org/docs/style-guide) and also follow the [Todo App](https://github.com/evancz/elm-todomvc/blob/master/Todo.elm).
- I changed `Edit.elm` so the coach related methods now receive `Coach` (not `ViewModel`) and the messages related to receive a `Message`. This makes things easier to have separated models in the future.

### CODE REVIEW

**Comment by viniciusffj** [comment]
@rodrigo-morais

File: src/Coaches/Edit/Effects.elm [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import Http
+import Task  
+import Effects  
+
+
+import Coaches.Edit.Actions exposing (..)
+import Coaches.Edit.Models exposing (..)
+
+
+savePlayers : Coach -> Effects.Effects Action
```
Comment (rodrigo-morais) [inline src/Coaches/Edit/Effects.elm]: I believe that the correct name is `saveCoaches`. Is it right?

File: src/Coaches/Edit/Effects.elm [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    , url = url
+    , body = body
+    }
+  in
+    Http.fromJson decoder (Http.send Http.defaultSettings request)
+
+
+saveUrl : String
+saveUrl =
+    "http://localhost:4000/coaches"
```
Comment (rodrigo-morais) [inline src/Coaches/Edit/Effects.elm:42]: What do you think if we put this address in a config file?

File: src/Coaches/Edit/Update.elm [hunk 3]
```diff
@@ -6,22 +6,66 @@ import Effects exposing (Effects)
...
-          let
-            updateMessage message =
-              { message | text = "Error when saving coach.", messageType = Error }
+    updateCoach newCoach = 
+      ({model | coach = newCoach}, Effects.none) 
+  in
+    case action of 
+      NoOp -> (model, Effects.none)
 
+      Save -> (model, savePlayers coach)
```
Comment (rodrigo-morais) [inline src/Coaches/Edit/Update.elm]: If you change the name of `savePlayers` in Effects file, you have to change here too.

File: src/Coaches/Edit/Update.elm [hunk 4]
```diff
@@ -6,22 +6,66 @@ import Effects exposing (Effects)
...
 
-            updatedViewModel =
-              { coach = model.coach, message = (updateMessage model.message) }
+      SaveDone result -> 
+        case result of
+          Ok coach ->
+            let
+              updatedViewModel =
+                { coach = model.coach
+                , message = successMessage
```
Comment (rodrigo-morais) [inline src/Coaches/Edit/Update.elm:31]: I really liked how you did it. :+1:

File: src/Coaches/Edit/Update.elm [hunk 5]
```diff
@@ -6,22 +6,66 @@ import Effects exposing (Effects)
...
+        updateCoach({coach | coach = value})
+
+      UpdateCapabilities value -> 
+        updateCoach({coach | capabilities = value})
+
+      UpdateDescription value -> 
+        updateCoach({coach | description = value})
+
+
+errorMessage : Message
```
Comment (rodrigo-morais) [inline src/Coaches/Edit/Update.elm:61]: Good call!

File: src/Coaches/Edit/Update.elm [hunk 6]
```diff
@@ -6,22 +6,66 @@ import Effects exposing (Effects)
...
+errorMessage =
+  { text = "Error when saving coach."
+  , messageType = Error 
+  }
+
+successMessage : Message
```
Comment (rodrigo-morais) [inline src/Coaches/Edit/Update.elm:67]: Good call!

File: src/Coaches/Edit/Edit.elm [hunk 7]
```diff
@@ -14,19 +13,19 @@ import Coaches.Edit.Actions exposing (..)
...
 view address model =
-  div []
-      [
-        flash address model,
-        nav address model,
-        form address model
-      ]
+  div 
+    []
+    [ flash address model.message
```
Comment (rodrigo-morais) [inline src/Coaches/Edit/Edit.elm:18]: I liked this approach to pass just what is necessary.
Cool!

**Comment by rodrigo-morais** [comment]
I don't like too much of some style options which Elm follows, it's a personal opinion because for me makes hard read the code. But I think we have to follow it and in some time I'll be accustomed with.

**Comment by viniciusffj** [comment]
@rodrigo-morais I've changed the wrong name.

I also don't like some styles, and I think the guide has just a little bit of information, but if the community is using, I think we should follow.

I didn't get the config file thing, the idea is to have a module with some "constants" and import it? Or to have a config file that we import in build time?

**Pull request closed without merging**